testutils/src/main/scala/za/co/absa/enceladus/testutils/models/TestRun.scala

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2018-2019 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.enceladus.testutils.models
+
+import org.joda.time.{DateTime, DateTimeZone}
+
+import scala.collection.mutable
+
+class TestRun(name: String) {
+  val startTime: DateTime = DateTime.now(DateTimeZone.UTC)
+  lazy val endTime: DateTime = DateTime.now(DateTimeZone.UTC)
```

Comment:
Why the `lazy val`? You only, use it in the `getFinalInfo` function. It could easily be a local variable and it won't need to be lazy.

Reply:
It just seems like a flawed design since the API allows me to set `endTime` at any point and unintentionally calling the getter produces a side-effect - setting the value.